parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -434,6 +440,63 @@ public static final ParquetMetadata readFooter(Configuration configuration, File
     }
   }
 
+  public static final Map<ColumnDescriptor, Dictionary> readDictionary(BlockMetaData blockMetadata,
+                                                                        Path path,
+                                                                        Configuration configuration,
+                                                                        MessageType requestedSchema) throws IOException
+  {
+    Map<ColumnDescriptor, Dictionary> dictionaries = new HashMap<ColumnDescriptor, Dictionary>();
+    for (ColumnChunkMetaData columnChunkMetaData : blockMetadata.getColumns()) {
+      for (ColumnDescriptor columnDescriptor : requestedSchema.getColumns()) {
+        if (columnChunkMetaData.getPath().equals(ColumnPath.get(columnDescriptor.getPath())) &&
+            isOnlyDictionaryEncodingPages(columnChunkMetaData.getEncodings())) {
+          long startingPosition = columnChunkMetaData.getStartingPos();
+          FSDataInputStream inputStream = path.getFileSystem(configuration).open(path);
+          inputStream.seek(startingPosition);
+
+          int totalSize = (int) columnChunkMetaData.getTotalSize();
+          byte[] buffer = new byte[totalSize];
+          inputStream.readFully(buffer);
+
+          DictionaryStream dictionaryStream = new DictionaryStream(buffer, 0);
+          DictionaryPage compressedPage = dictionaryStream.readDictionaryPage();
+          inputStream.close();
+
+          if (compressedPage == null) {
+            dictionaries.put(columnDescriptor, null);
+            break;
+          }
+
+          CodecFactory codecFactory = new CodecFactory(configuration);
+          BytesDecompressor decompressor = codecFactory.getDecompressor(columnChunkMetaData.getCodec());
+          DictionaryPage dictionaryPage = new DictionaryPage(decompressor.decompress(compressedPage.getBytes(),
+                                                                                    compressedPage.getUncompressedSize()),
+                                                            compressedPage.getDictionarySize(),
+                                                            compressedPage.getEncoding());
+          Dictionary dictionary = dictionaryPage.getEncoding().initDictionary(columnDescriptor, dictionaryPage);
+          dictionaries.put(columnDescriptor, dictionary);
+          break;
+        }
+      }
+    }
+    return dictionaries;
+  }
+
+  private static boolean isOnlyDictionaryEncodingPages(Set<Encoding> encodings) {
+    // more than 1 encodings for values
+    if (encodings.size() > PARQUET_DATA_TRIPLE) {
+      return false;
+    }
```

Comment:
please verify your heuristics. There could be some other cases.
PAGE_V2 formalizes this contract.